modin/core/io/sql/connection.py

```diff
@@ -0,0 +1,74 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""
+Module houses `ModinDatabaseConnection` class.
+
+`ModinDatabaseConnection` lets a single process make its own connection to a
+database to read from it. Whereas it's possible in pandas to pass an open
+connection directly to `read_sql`, the open connection is not pickleable
+in Modin, so each worker must open its own connection.
+`ModinDatabaseConnection` saves the arguments that would normally be used to
+make a db connection. It can make and provide the a connection whenever the
+Modin driver or a worker wants one.
+"""
+
+_PSYCOPG_LIB_NAME = "psycopg2"
+_SQLALCHEMY_LIB_NAME = "sqlalchemy"
+
+
+class ModinDatabaseConnection:
```

Comment:
A top-level API would be a bit better. 

```python
from modin.db_conn import ModinDatabaseConnection
```

Other things will go in that `db_conn` folder like SQL statement construction for specific SQL engines (Like MS SQL) and it feels more user-friendly. What do you think?

Reply:
I agree. Done.